js/reference.js

```diff
@@ -52,23 +55,36 @@ var ERMrest = (function(module) {
      * {@link ERMrest.Unauthorized},
      * {@link ERMrest.NotFoundError},
      */
-    module.resolve = function(uri) {
+    function resolve(ermrestUri) {
         try {
+            var uri = ermrestUri.baseUri + ermrestUri.hash;
             verify(uri, "'uri' must be specified");
-            // TODO
-            // parse the uri; validating its syntax here
-            //  if invalid syntax; reject with malformed uri error
-            // make a uri to the catalog schema resource
-            // get the catalog/N/schema
-            // validate the model references in the `uri` parameter
-            // this method needs to internally construct a reference object that
-            // represents the `uri` parameter
-            notimplemented();
+
+            var defer = module._q.defer();
+
+            // build reference
+            var context = module._parse(uri);
+            context.baseUri = ermrestUri.baseUri;
+            var reference = new Reference(context);
+
+            var server = this.getServer(reference._serviceUrl);
+            server.catalogs.get(reference._catalogId).then(function success(catalog) {
+                // Should I make a setter here?
+                reference._catalog = catalog;
+                var schema = reference._schema = catalog.schemas.get(reference._schemaName);
+                reference.table = schema.tables.get(reference._tableName);
```

Comment:
i don't think we specified that `reference` would have a public `.table` property did we?